src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -200,8 +240,37 @@ private static void OnAvailabilityTimerFired(object state)
 
             if (changed)
             {
-                s_availabilityChangedSubscribers?.Invoke(null, new NetworkAvailabilityEventArgs(NetworkInterface.GetIsNetworkAvailable()));
+                var isAvailable = NetworkInterface.GetIsNetworkAvailable();
+                foreach (var subscriber in s_availabilityChangedSubscribers)
+                {
+                    NetworkAvailabilityChangedEventHandler handler = subscriber.Key;
+                    ExecutionContext ec = subscriber.Value;
+
+                    if (ec == null) // Flow supressed
+                    {
+                        handler(null, isAvailable ? s_availableEventArgs : s_notAvailableEventArgs);
```

Comment:
Can you move these `isAvailable ? a : b` checks to be before the loop, rather than doing it for each invocation?

Reply:
Done